lib/commands/help.js

```diff
@@ -61,9 +61,10 @@ class Help extends BaseCommand {
     const f = `${manroot}/${manSearch}/?(npm-)${section}.[0-9]*`
     let mans = await glob(f)
     mans = mans.sort((a, b) => {
-      // Because of the glob we know the manNumberRegex will pass
-      const aManNumber = a.match(manNumberRegex)[1]
-      const bManNumber = b.match(manNumberRegex)[1]
+      // Because the glob is (subtly) different from manNumberRegex,
+      // we can't rely on it passing.
+      const aManNumber = (a.match(manNumberRegex) || Infinity)[1]
```

Comment:
I don’t think you want index 1 of Infinity
```suggestion
      const aManNumber = a.match(manNumberRegex)?.[1] || Infinity
```
(assuming optional chaining is available in all of npm’s supported nodes)

Reply:
It is not

Reply:
Then the match object probably needs to be saved in its own variable, and the [1] done separately.

Reply:
Fixed. I'm not sure how that worked in my manual testing...

Reply:
you tested in a version of node that supported it. npm has to run in older versions that don't have that feature.